webapp/src/app/core/types/classes.ts

```diff
@@ -25,6 +26,23 @@ export class BlockData {
     return this.type === BlockType.Input;
   }
 
+  check(validation: BlockValidation, isChecked: boolean): void {
```

Comment:
How about `setIsChecked`? `check` sounds like it might also be a boolean method to check something.

`isChecked` could then be `getIsChecked`.